packages/mgt-chat/src/statefulClient/BotInfoClient.ts

```diff
@@ -0,0 +1,50 @@
+import { TeamsAppInstallation } from '@microsoft/microsoft-graph-types-beta';
+import { BaseStatefulClient } from './BaseStatefulClient';
+import { loadBotInChat, loadBotIcon } from './graph.chat';
+
+export interface BotInfo {
+  botInfo: Map<string, TeamsAppInstallation>;
+  botIcons: Map<string, string>;
+  loadBotInfo: (chatId: string, appId: string) => Promise<void>;
+}
+
+const requestKey = (chatId: string, appId: string) => `${chatId}::${appId}`;
+
+export class BotInfoClient extends BaseStatefulClient<BotInfo> {
+  private readonly infoRequestMap = new Map<string, Promise<unknown>>();
+
+  public loadBotInfo = async (chatId: string, appId: string) => {
+    const beta = this.betaGraph;
+    if (!beta) return;
+    const key = requestKey(chatId, appId);
+    if (!this.infoRequestMap.has(key)) {
+      const requestPromise = loadBotInChat(beta, chatId, appId);
+      this.infoRequestMap.set(key, requestPromise);
+      const botInfo = await requestPromise;
+      if (botInfo?.value.length > 0) {
+        // update state with the text info
+        this.notifyStateChange((draft: BotInfo) => {
+          botInfo.value.forEach(app => {
+            if (app.teamsAppDefinition?.bot?.id) {
+              draft.botInfo.set(app.teamsAppDefinition?.bot?.id, app);
+            }
+          });
+        });
+        // load the appIcon for each bot
+        for (const app of botInfo.value) {
+          const appIcon = await loadBotIcon(beta, app);
```

Comment:
consider whether you want these to be loaded serially or in parallel.

Reply:
given the nature of the filter clause to load the data about the installed app there should only be one item in the value collection.

If there were multiple I'd be putting in the work to do this in parallel, but given this use case I don't see the benefit.

Reply:
ah yes, I forgot you were loading by ID.